### PR TITLE
Bug Fixing - Path Smoothing failures

### PR DESCRIPTION
**Pull request by Ulvetanna**
# PolarRoute Pull Request Template

Date:    
Version Number:    
 
## Description of change

Fixing bug raised in #303 

Fixed bug where dijkstra was called multiple times for the same graph

Fixed bug where early stopping criterion was not present in the codebase

## Fixes # (issue)

# Testing
To ensure that the functionality of the PolarRoute codebase remains consistent throughout the development cycle a testing strategy has been developed, which can be viewed in the document `test/testing_strategy.md`. 
This includes a collection of test files which should be run according to which part of the codebase has been altered in a pull request. Please consult the testing strategy to determine which tests need to be run. 

- [ ] My changes have not altered any of the files listed in the testing strategy

- [ ] My changes result in all required regression tests passing without the need to update test files.  
  
> *list which files have been altered and include a pytest.txt file for each of
> the tests required to be run*
>
> The files which have been changed during this PR can be listed using the command

    git diff --name-only 1.1.x

- [ ] My changes require one or more test files to be updated for all regression tests to pass.   

> *include pytest.txt file showing which tests fail.*  
> *include reasoning as to why your changes cause these tests to fail.* 
>
> Should these changes be valid, relevant test files should be updated.  
> *include pytest.txt file of test passing after test files have been updated.*

# Checklist

- [ ] My code follows [pep8](https://peps.python.org/pep-0008/) style guidelines.  
- [x] I have commented my code, particularly in hard-to-understand areas.  
- [x] I have updated the documentation of the codebase where required.  
- [x] My changes generate no new warnings.   
- [x] My PR has been made to the `1.1.x` branch (**DO NOT SUBMIT A PR TO MAIN**)

### CODE REVIEW

**Comment by Ulvetanna** [comment]
[regression_tests.txt](https://github.com/user-attachments/files/17585940/regression_tests.txt)

Adding in regression tests

File: polar_route/route_planner/route_planner.py [hunk 1]
```diff
@@ -301,11 +309,15 @@ def _splitting_around_waypoints(self, waypoints_df):
         """
         if ('waypoint_splitting' in self.config) and (self.config['waypoint_splitting']):
             logging.info(' Splitting around waypoints !')
+            prior_cells = [cellbox.get_id() for cellbox in self.env_mesh.agg_cellboxes if cellbox.agg_data['inaccessible']]
```
Comment (gecoombs) [inline polar_route/route_planner/route_planner.py:312]: Should this be `if not cellbox.agg_data['inaccessible']`? (same on line 315 below)

Comment (Ulvetanna) [reply polar_route/route_planner/route_planner.py:312]: good spot, that shouldn't be the case. Will correct now and rerun tests now.

**Review by gecoombs — CHANGES_REQUESTED** [review]
Looks good, apart from my previous comment, could you also update the version number?

File: polar_route/route_planner/route_planner.py (hunk 1 above)
Comment (Ulvetanna) [reply polar_route/route_planner/route_planner.py]: Fixed, and re-ran the regression tests. 
[regression_tests.txt](https://github.com/user-attachments/files/17586587/regression_tests.txt)

**Review by gecoombs — APPROVED** [review]
Seems legit 👍

**Pull request merged**